hello.html

```diff
@@ -7,83 +7,146 @@
+         * ***************************************************************/
+        _bridge.init = function(_jsbridge) {
+          if (BRIDGE_ON) {
+            _jsbridge.init(function(message, responseCallback) {
+              console.log('Received message: ' + message);
+            });
+
+            jsbridge = _jsbridge;
+          }
+        };
+
+        /** **************************************************************
+         * Sets up JS Bridge without reinitializing.
+         * @param {WebViewJavascriptBridge} _jsbridge the bridge
+         * ***************************************************************/
+        _bridge.setup = function(_jsbridge) {
+          jsbridge = _jsbridge;
+        }
 
-            return _bridge;
-        }();
+        return _bridge;
+      }();
+      /* ******************************
+       *  END Pulsar bridge definition
+       * ******************************/
+
+      /* ******************************
+       * BEGIN Pulsar onload definition
+       * ******************************/
+      const runningEmbedded = (window.top !== window.self);
+      window.onload = function() {
+
+        if (runningEmbedded && window.parent.pulsar && window.parent.pulsar.bridge) {
+          /* ************************************************************
+           * If the above is true, we are running in an embedded state
+           * within the Pulsar for FSL app. In this case, the above
+           * registered event listener will never fire as the webview we
+           * are running in already has established a Pulsar JS bridge.
+           * ************************************************************/
+          window.pulsar = window.parent.pulsar; // ensure we will pass down the embedded window.pulsar
+          console.assert(window.pulsar.bridge !== undefined);
+          bridge.setup(window.pulsar.bridge);
+
+          /* At this point we have the correct Pulsar JS Bridge object from the window itself.
+           * Start your app! */
+          startApp();
+
+        } else {
+          if (runningEmbedded) {
+            /* ************************************************************
+             * Something has gone wrong if we are running embedded and find
+             * ourselves without a window.parent.pulsar.bridge or
+             * window.parent.pulsar object.
+             * ************************************************************/
+            console.log('OOPS, running embedded, but no window.parent.pulsar.bridge!');
+            throw 'embedded misconfiguration!';
+          } else {
+            /* ************************************************************
+             * Otherwise we are running in a stand-alone state in our own
+             * webview context. The Pulsar JS bridge will be established by
+             * the event listener above. So we do nothing here.
+             * ************************************************************/
+          }
+        }
```

Comment:
I think it would be useful to have the prior `console.assert()` and `console.log()` code here for our developers to see.